opte/src/engine/packet.rs

```diff
@@ -1919,8 +1891,17 @@ pub enum ReadErr {
 pub enum WriteError {
     BadLayout,
     EndOfPacket,
-    NotEnoughBytes { available: usize, needed: usize },
-    OutOfRange,
+    NotEnoughBytes {
+        start: *const u8,
+        end: *const u8,
+        available: usize,
+        needed: usize,
+    },
+    OutOfRange {
+        start: *const u8,
+        limit: *const u8,
```

Comment:
IIUC, the field name `limit` comes from the `db_lim` field of the `struct datab`, which is described as the "last byte (+1) of buffer". My reading is that this is the first byte _past the end_ of the buffer, so that the size of the buffer is `db_lim - db_base`. If that's all correct, I think this name could use a quick comment, which could be just a reference to the `db_lim` field and/or `man datab.9s`.

Reply:
This field isn't referring to `db_lim`, it's the end of the buffer that the write has available to it. In the case of a `Modify` that is `wptr`, as it can only write over existing bytes. I pushed a change that hopefully makes these field names more clear.